### PR TITLE
Bugfix/vulnabilitytrackinguri Tightened URLpattern validation and added comprehensive unit tests

### LINKED ISSUE #323: [Issue]: vulnerabilityTrackingURI field must match format "URL" error prevents document opening
### Issue Description

When opening an ISRA xml containing a the element `<my:vulnerabilityTrackingURI>http:///</my:vulnerabilityTrackingURI>` results in a dialog to pop up. Since this error always happens on opening the file, there is no way to use the tool with the file. 

Below is the screenshot of the observed error:

Image

**Workaround**
Manually update the XML to fix the URI

### Duplicate Steps

1. Take an existing XML
2. Find one occurrence of `<my:vulnerabilityTrackingURI></my:vulnerabilityTrackingURI>`, edit it to `<my:vulnerabilityTrackingURI>http:///</my:vulnerabilityTrackingURI>`
3. Save the XML
4. Open the XML in sratool

### Expected Results

The tool either removes the invalid value or provides a temporary valid value. At the end, the user must be able to fix the issue within the tool, so it must be possible to open and edit the file in sratool.

### Version Detected

1.2.0

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by DylanAwThales**
### Description
This PR updates the URL validation regex (URLpattern) and adds a dedicated Jest test suite to lock in expected behaviour across supported URI schemes. It also updates existing test fixtures that no longer match the stricter rules.

**Additional Information**
URLpattern is used in two critical places:

Schema validation during import via AJV

Runtime validation in class setters
Without strong automated coverage, regex changes can silently break imports, edits, or existing fixtures. This PR makes the rules explicit and prevents regressions.

### Changes

Updated URLpattern implementation
File: lib/src/model/schema/validation-pattern/validation-pattern.js
Refined URLpattern to reflect the new accepted and rejected formats.

Added new unit test suite for URLpattern
File: lib/test/unit/validation-pattern/urlpattern.test.js
Covers http and https, ftp, mailto, tel, urn, empty string, plus a wide set of invalid edge cases.

Updated existing fixtures affected by stricter validation
Files: lib/test/unit/test-8/test-8.test.js
Adjusted samples (for example ftp fixtures) so they match the updated regex and do not fail setter validation.

Fixes #323

### CODE REVIEW

File: lib/src/api/xml-json/alter-isra/alter-vulnerability.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 /*----------------------------------------------------------------------------
 *
-*     Copyright © 2022 THALES. All Rights Reserved.
- *
+*     Copyright © 2025 THALES. All Rights Reserved.
```
Comment (Alvin-JohnPagente) [inline lib/src/api/xml-json/alter-isra/alter-vulnerability.js]: Should be 2022-2025

**Review by Alvin-JohnPagente — CHANGES_REQUESTED** [review]
- This PR contains commits that is not related to the bug. Remove those commits.
- I need you to explain what is the cause of the issue and what happens when the URL is invalid
